### PR TITLE
Add note about native-set EVs

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #39448

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md](https://github.com/dotnet/docs/blob/6c6b774496e7e8f20307e0566bd42cab4344a50b/docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md) | [System.Environment.GetEnvironmentVariable methods](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-environment-getenvironmentvariable?branch=pr-en-us-40029) |

### CODE REVIEW

File: docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md [hunk 1]
```diff
@@ -9,14 +9,17 @@ ms.date: 01/24/2024
 
 The <xref:System.Environment.GetEnvironmentVariable%2A> method retrieves the value of an environment variable from the current process.
 
+Environment variable names are case-sensitive on Unix-like systems but aren't case-sensitive on Windows.
+
+> [!NOTE]
+> In-process environment modifications made by native libraries aren't seen by managed callers.
```
Comment (mklement0) [inline docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md:15]: Great, thanks for tackling this too. It  may make sense to cover _both_ directions here, by adding something like: "Conversely, such modifications made by managed callers aren't seen by native libraries."

File: docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md [hunk 2]
```diff
@@ -29,29 +32,27 @@ On Windows systems, the environment block of the current process includes:
 
 If environment variables are created after the process has started, you can use this method to retrieve only those variables that were created by calling the <xref:System.Environment.SetEnvironmentVariable(System.String,System.String)> method or the <xref:System.Environment.SetEnvironmentVariable(System.String,System.String,System.EnvironmentVariableTarget)> method with a `target` value of .<xref:System.EnvironmentVariableTarget.Process?displayProperty=nameWithType>.
 
-### On non-Windows systems
+### On Unix-like systems
 
-On non-Windows systems, the environment block of the current process includes the following environment variables:
+On Unix-like systems, the environment block of the current process includes the following environment variables:
```
Comment (BillWagner) [inline docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md:37]: Is it worth mentioning that macOS is "unix-like", or is that obvious?

**Review by BillWagner — APPROVED** [review]
This looks great @gewarren 

Once the comments are addressed, let's :shipit:

File: docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md (hunk 2 above)
Comment (gewarren) [reply docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md:37]: We use "Unix-like" throughout the docs, so I think this is okay.

File: docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md (hunk 1 above)
Comment (gewarren) [inline docs/fundamentals/runtime-libraries/system-environment-getenvironmentvariable.md:15]:
```suggestion
> In-process environment modifications made by native libraries aren't seen by managed callers. Conversely, such modifications made by managed callers aren't seen by native libraries.
```

**Pull request merged**